src/engine/x86-64/X86_64SinglePassCompiler.v3

```diff
@@ -552,7 +552,7 @@ class X86_64SinglePassCompiler extends SinglePassCompiler {
 	def visit_F64X2_SUB() { do_op2_x_x(ValueKind.V128, asm.subpd_s_s); }
 	def visit_F64X2_MUL() { do_op2_x_x(ValueKind.V128, asm.mulpd_s_s); }
 	def visit_F64X2_DIV() { do_op2_x_x(ValueKind.V128, asm.divpd_s_s); }
-	def visit_F64X2_NEG() { visit_V128_F_NEG_ABS(mmasm.emit_v128_negpd); }
+	def visit_F64X2_NEG() { do_op1_x(ValueKind.V128, mmasm.emit_v128_negpd(_, G(allocTmp(ValueKind.I64)), X(allocTmp(ValueKind.V128)))); }
```

Comment:
What about factoring out a routine that does the temporary allocation?

Reply:
Yeah that makes things more clear. Updates pushed.